File: sdk/src/main/java/io/antelli/sdk/model/Answer.java
package io.antelli.sdk.model;

import android.content.Intent;
import android.os.Bundle;
import android.os.Parcel;
import android.os.Parcelable;

import java.util.ArrayList;
import java.util.List;

/**
 * Handcrafted by Štěpán Šonský on 27.08.2017.
 */

public class Answer implements Parcelable {

    private static final String PARAM_AUTO_LISTEN = "AUTO_LISTEN";
    private static final String PARAM_AUTO_RUN = "AUTO_RUN";

    private List<AnswerItem> items = new ArrayList<>();
    private List<Hint> hints;
    private Bundle params = new Bundle();

    public Answer() {
    }

    public Answer(String text) {
        addItem(new AnswerItem().setText(text).setSpeech(text));
    }

    public Answer(List<AnswerItem> items) {
        this.items.addAll(items);
    }

    public Answer addItem(AnswerItem item) {
        this.items.add(item);
        return this;
    }

    public Answer addItems(List<AnswerItem> item) {
        this.items.addAll(item);
        return this;
    }

    public boolean isAutoListen() {
        return params.getBoolean(PARAM_AUTO_LISTEN, false);
    }

    public Answer setAutoListen(boolean autoListen) {
        params.putBoolean(PARAM_AUTO_LISTEN, autoListen);
        return this;
    }

    public void addHint(Hint hint) {
        if (hints == null) {
            hints = new ArrayList<>();
        }
        hints.add(hint);
    }

    public Answer setHints(List<Hint> hints){
        this.hints = hints;
        return this;
    }

    public Answer setAutoRun(Intent intent){
        params.putParcelable(PARAM_AUTO_RUN, intent);
        return this;
    }

    public Intent getAutoRun(){
        return params.getParcelable(PARAM_AUTO_RUN);
    }

    public List<AnswerItem> getItems() {
        return items;
    }

    public List<Hint> getHints() {
        return hints;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeTypedList(this.items);
        dest.writeTypedList(this.hints);
        dest.writeBundle(this.params);
    }

    protected Answer(Parcel in) {
        this.items = in.createTypedArrayList(AnswerItem.CREATOR);
        this.hints = in.createTypedArrayList(Hint.CREATOR);
        this.params = in.readBundle(getClass().getClassLoader());
    }

    public static final Creator<Answer> CREATOR = new Creator<Answer>() {
        @Override
        public Answer createFromParcel(Parcel source) {
            return new Answer(source);
        }

        @Override
        public Answer[] newArray(int size) {
            return new Answer[size];
        }
    };
}

File: sdk/src/main/java/io/antelli/sdk/model/Answer.kt
package io.antelli.sdk.model

import android.content.Intent
import android.os.Bundle
import android.os.Parcel
import android.os.Parcelable
import java.util.*
import kotlin.collections.ArrayList

/**
 * Handcrafted by Štěpán Šonský on 27.08.2017.
 */
open class Answer : Parcelable {
    var items: MutableList<AnswerItem> = ArrayList()
        private set
    private var params = Bundle()

    constructor()
    constructor(text: String?) {
        addItem(AnswerItem().apply {
            this.text = text
            this.speech = text
        })
    }

    constructor(items: List<AnswerItem>) {
        this.items.addAll(items)
    }

    fun addItem(item: AnswerItem) {
        items.add(item)
    }

    fun addItems(item: List<AnswerItem>) {
        items.addAll(item)
    }

    var isAutoListen: Boolean
        get() = params.getBoolean(PARAM_AUTO_LISTEN, false)
        set(value) = params.putBoolean(PARAM_AUTO_LISTEN, value)

    var autoRun: Intent?
        get() = params.getParcelable(PARAM_AUTO_RUN)
        set(value) = params.putParcelable(PARAM_AUTO_RUN, value)

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeTypedList(items)
        dest.writeBundle(params)
    }

    protected constructor(`in`: Parcel) {
        items = `in`.createTypedArrayList(AnswerItem.CREATOR) ?: ArrayList()
        params = `in`.readBundle(javaClass.classLoader) ?: Bundle()
    }

    companion object {
        private const val PARAM_AUTO_LISTEN = "AUTO_LISTEN"
        private const val PARAM_AUTO_RUN = "AUTO_RUN"

        @JvmField
        val CREATOR: Parcelable.Creator<Answer> = object : Parcelable.Creator<Answer> {
            override fun createFromParcel(source: Parcel): Answer? {
                return Answer(source)
            }

            override fun newArray(size: Int): Array<Answer?> {
                return arrayOfNulls(size)
            }
        }
    }
}